drivers/wifi/nrf700x/src/wifi_mgmt.c

```diff
@@ -670,3 +670,169 @@ void nrf_wifi_event_proc_twt_sleep_zep(void *vif_ctx,
 	break;
 	}
 }
+
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
+int nrf_wifi_mode(const struct device *dev,
+		  struct wifi_mode_info *mode)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct nrf_wifi_ctx_zep *rpu_ctx_zep = NULL;
+	struct nrf_wifi_vif_ctx_zep *vif_ctx_zep = NULL;
+	struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx = NULL;
+	struct nrf_wifi_fmac_dev_ctx_def *def_dev_ctx = NULL;
+	int ret = -1;
+
+	if (!dev || !mode) {
+		goto out;
+	}
+	vif_ctx_zep = dev->data;
+
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL", __func__);
+		goto out;
+	}
+
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
+	def_dev_ctx = wifi_dev_priv(fmac_dev_ctx);
+
+	if (!device_is_ready(dev)) {
+		LOG_ERR("%s: Device %s is not ready",
+			__func__, dev->name);
+		goto out;
+	}
+
+	if (mode->oper == WIFI_MGMT_SET) {
+		status = nrf_wifi_check_mode_validity(mode->mode);
+		if (status != NRF_WIFI_STATUS_SUCCESS) {
+			goto out;
+		}
+
+		/**
+		 * Send the driver vif_idx instead of upper layer sent if_index.
+		 * we map network if_index 1 to vif_idx of 0 and so on. The vif_ctx_zep
+		 * context maps the correct network interface index to current driver
+		 * interface index.
+		 */
+		status = nrf_wifi_fmac_mode(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, mode->mode);
+		if (status != NRF_WIFI_STATUS_SUCCESS) {
+			LOG_ERR("%s: filter operation failed", __func__);
+			goto out;
+		}
+
+	} else {
+		mode->mode = def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->mode;
+	}
+	ret = 0;
+out:
+	return ret;
+}
+
+int nrf_wifi_filter(const struct device *dev,
```

Comment:
Should not be protected with `#ifdef CONFIG_NRF700X_RAW_DATA_TX`

Reply:
okay